relay-server/src/utils/thread_pool.rs

```diff
@@ -63,18 +65,74 @@ impl ThreadPoolBuilder {
     }
 }
 
+/// A [`WorkerGroup`] adds an async brackpressure mechanism to a [`ThreadPool`].
```

Comment:
Why do we need this backpressure mechanism? The old processor required it because the thread pool was larger than the amount of processing jobs we wanted to run at the same time. But with a dedicated thread pool, there is already a maximum number of parallel tasks.

That is, by using a `WorkerGroup` instead of a `ThreadPool` in `store.rs`, the backlog shifts from a ThreadPool-internal queue to the store service queue. Does that bring us any advantages?

Reply:
Without it, we start draining the async/tokio channel and instead queue everything on the thread pool, which breaks our monitoring of the service channel, other than that, I don't think it'd be necessary.